mautic/mautic
# Dnc segment sql optimization

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL |  N
| Related developer documentation PR URL | N
| Issues addressed (#s or URLs) | 
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
SQL optimization for DNC segment. 

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Run segments:update command, SQL query described in addressed issue is running slow
2. SQL query is using apostrophes for reason parameter

#### Steps to test this PR:
1. Load up this PR
2. Check indexes in lead_donotcontact table
3. Run migrations:migrate
4. Index for lead_id has been deleted, new index for lead_id, channel, reason has been added
5. SQL query is not using apostrophes for reason parameter
